serde-reflection/tests/serde.rs

```diff
@@ -502,3 +502,26 @@ fn test_default_value_for_primitive_types() {
     assert_eq!(format, Format::Str);
     assert_eq!(value, "A borrowed str");
 }
+
+#[test]
+fn test_trace_value() {
+    let mut tracer = Tracer::new(TracerConfig::default());
+
+    #[derive(Serialize, Deserialize)]
+    struct Thing {
+        value: serde_json::Value,
+        other_field: u32,
+    }
+
+    tracer.trace_simple_type::<Thing>().unwrap();
+
+    let registry = tracer.registry().unwrap();
+    match registry.get("Thing").unwrap() {
+        ContainerFormat::Struct(fields) => {
+            assert_eq!(fields.len(), 2);
+            assert_eq!(fields[0].value, Format::Any);
```

Comment:
With a little bit of boilerplate in user code, we could have `Format::Typename` here.

Specifically, there could be a wrapper struct around `serde_json::Value` (say `struct Value(serde_json::Value`)) so that the (de)serialize trait implementations are treating the type as a plain enum for non-human-readable 
(de)serializers.

Reply:
Or alternatively use `#[serde(with = "module")]` instead of a wrapper type ([doc](https://serde.rs/field-attrs.html)).

Here is an example of conditional implementation of (de)serialize traits:
https://github.com/linera-io/linera-protocol/blob/152ebf5fb767ae967b381d3ed3b64dd9aedc8d9e/linera-base/src/data_types.rs#L60

Reply:
If I understand correctly, this would indeed work perfectly. The problem is that the generated XML (which is the serde data format I need to use, sadly :p)  would be less readable, explicitly encoding whether each field is a Value::Int, Value::String, etc. You'd get XML like this
```
<person>
  <object>
      <name><string>Jonathan</string></name>
      <age><number>23</number></age>
  </object>
</person>
```
Instead of
```
<person>
    <name>Jonathan</name>
    <age>23</age>
</person>
```

To clarify my goal, I'm building a new serde data format crate which can generate XML and outputs a corresponding XSD using the serde-reflection crate.

Reply:
> Or alternatively use `#[serde(with = "module")]` instead of a wrapper type ([doc](https://serde.rs/field-attrs.html)).
> 
> Here is an example of conditional implementation of (de)serialize traits: https://github.com/linera-io/linera-protocol/blob/152ebf5fb767ae967b381d3ed3b64dd9aedc8d9e/linera-base/src/data_types.rs#L60

Oh I see, with conditional implementation of serialize/deserialize this could indeed work. Bit of a hacky solution, but I could try doing it that way

Reply:
I understand that replacing serde_json::Value or annotating the fields could be annoying but for the rest what prevents you from obtaining the desired generated output once you have a format with a recognizable typename?

Reply:
Currently my code was hitting the Err::NotSupported case in this crate's deserialize implementation, therefore I'm not getting a Registry at all. The example you sent above where a serialize/deserialize is switching its implementation based on the is_human_readable flag would be a workaround for this problem, so the deserialize_any call is not triggered and then I could manually filter out the types that use deserialize_any. This is a bit less nice to use than the implementation in this PR but an acceptable compromise

Reply:
Yes obviously you can pick the solution that works for you. Given that there is a workaround, though, I'm leaning towards keeping the codebase as it is.

Reply:
That's fair enough, then I'll close this PR, thanks a lot for the review!